### PR TITLE
Fix broken links in documentation

### PR DESCRIPTION
**Pull request by user27182**
### Overview

Fix a number of errors discovered by #6895

<details>

``` yaml

2024-11-21T19:28:52.6073205Z /home/runner/work/pyvista/pyvista/pyvista/core/filters/data_set.py:docstring of pyvista.core.filters.data_set.DataSetFilters.extract_cells_by_type:10: WARNING: py:class reference target not found: pyvista.RectilnearGrid [ref.class]
2024-11-21T19:28:52.6930810Z /home/runner/work/pyvista/pyvista/doc/source/examples/01-filter/interpolate_sample.rst:300: WARNING: py:func reference target not found: pyvista.DataSetFilters.sample` [ref.func]
2024-11-21T19:28:52.6932842Z /home/runner/work/pyvista/pyvista/doc/source/examples/99-advanced/fea-hertzian-contact-pressure.rst:64: WARNING: py:mod reference target not found: pyvista.examples [ref.mod]
2024-11-21T19:28:52.6935335Z /home/runner/work/pyvista/pyvista/doc/source/examples/99-advanced/magnetic-fields.rst:29: WARNING: py:func reference target not found: pyvista.PolyDataFilters.streamlines_from_source [ref.func]
2024-11-21T19:28:52.6937151Z /home/runner/work/pyvista/pyvista/CONTRIBUTING.rst:1033: WARNING: py:func reference target not found: pyvista.examples.download_guitar [ref.func]
2024-11-21T19:28:52.6938703Z /home/runner/work/pyvista/pyvista/CONTRIBUTING.rst:1034: WARNING: py:func reference target not found: pyvista.examples.download_trumpet [ref.func]
2024-11-21T19:28:52.6912857Z /home/runner/work/pyvista/pyvista/doc/source/api/utilities/_autosummary/pyvista.read_grdecl.rst:8: WARNING: py:attr reference target not found: pyvista.ExplicitStructuredGrid.user_dict [ref.attr]
2024-11-21T19:28:52.6805691Z /home/runner/work/pyvista/pyvista/pyvista/core/utilities/geometric_objects.py:docstring of pyvista.core.utilities.geometric_objects.Text3D:9: WARNING: py:attr reference target not found: center [ref.attr]
2024-11-21T19:28:52.6807740Z /home/runner/work/pyvista/pyvista/pyvista/core/utilities/geometric_objects.py:docstring of pyvista.core.utilities.geometric_objects.Text3D:13: WARNING: py:attr reference target not found: height [ref.attr]
2024-11-21T19:28:52.6809659Z /home/runner/work/pyvista/pyvista/pyvista/core/utilities/geometric_objects.py:docstring of pyvista.core.utilities.geometric_objects.Text3D:24: WARNING: py:attr reference target not found: height [ref.attr]
2024-11-21T19:28:52.6811551Z /home/runner/work/pyvista/pyvista/pyvista/core/utilities/geometric_objects.py:docstring of pyvista.core.utilities.geometric_objects.Text3D:30: WARNING: py:attr reference target not found: width [ref.attr]
2024-11-21T19:28:52.6813435Z /home/runner/work/pyvista/pyvista/pyvista/core/utilities/geometric_objects.py:docstring of pyvista.core.utilities.geometric_objects.Text3D:42: WARNING: py:attr reference target not found: depth [ref.attr]
2024-11-21T19:28:52.6884199Z /home/runner/work/pyvista/pyvista/doc/source/api/utilities/_autosummary/pyvista.fit_plane_to_points.rst:99: WARNING: py:meth reference target not found: pyvista.DataSetFilter.clip [ref.meth]
2024-11-21T19:28:52.6738496Z /home/runner/work/pyvista/pyvista/pyvista/plotting/renderer.py:docstring of pyvista.plotting.renderer.Renderer.add_bounding_box:63: WARNING: py:obj reference target not found: pyvista.DataSetFilter.bounding_box [ref.obj]
2024-11-21T19:28:52.6129377Z /home/runner/work/pyvista/pyvista/pyvista/core/filters/data_set.py:docstring of pyvista.core.filters.data_set.DataSetFilters.split_values:9: WARNING: py:meth reference target not found: pyvista.DataSetFilter.extract_values [ref.meth]
2024-11-21T19:28:52.6136490Z /home/runner/work/pyvista/pyvista/pyvista/core/filters/data_set.py:docstring of pyvista.core.filters.data_set.DataSetFilters.split_values:65: WARNING: py:meth reference target not found: pyvista.DataSetFilter.extract_values [ref.meth]
2024-11-21T19:28:52.6269128Z /home/runner/work/pyvista/pyvista/pyvista/core/filters/data_set.py:docstring of pyvista.core.filters.data_set.DataSetFilters.threshold:100: WARNING: py:obj reference target not found: extract_values [ref.obj]
2024-11-21T19:28:52.6481449Z /home/runner/work/pyvista/pyvista/pyvista/core/_validation/validate.py:docstring of pyvista.core._validation.validate.validate_array:19: WARNING: py:func reference target not found: pyvista.core.validation.check.check_has_shape [ref.func]
2024-11-21T19:28:52.6487173Z /home/runner/work/pyvista/pyvista/pyvista/core/_validation/validate.py:docstring of pyvista.core._validation.validate.validate_array:35: WARNING: py:func reference target not found: pyvista.core.validation.check.check_subdtype [ref.func]
2024-11-21T19:28:52.6491093Z /home/runner/work/pyvista/pyvista/pyvista/core/_validation/validate.py:docstring of pyvista.core._validation.validate.validate_array:42: WARNING: py:func reference target not found: pyvista.core.validation.check.check_has_length [ref.func]
2024-11-21T19:28:52.6493198Z /home/runner/work/pyvista/pyvista/pyvista/core/_validation/validate.py:docstring of pyvista.core._validation.validate.validate_array:55: WARNING: py:func reference target not found: pyvista.core.validation.check.check_has_length [ref.func]
2024-11-21T19:28:52.6495469Z /home/runner/work/pyvista/pyvista/pyvista/core/_validation/validate.py:docstring of pyvista.core._validation.validate.validate_array:59: WARNING: py:func reference target not found: pyvista.core.validation.check.check_has_length [ref.func]
2024-11-21T19:28:52.6497583Z /home/runner/work/pyvista/pyvista/pyvista/core/_validation/validate.py:docstring of pyvista.core._validation.validate.validate_array:63: WARNING: py:func reference target not found: pyvista.core.validation.check.check_nonnegative [ref.func]
2024-11-21T19:28:52.6499680Z /home/runner/work/pyvista/pyvista/pyvista/core/_validation/validate.py:docstring of pyvista.core._validation.validate.validate_array:67: WARNING: py:func reference target not found: pyvista.core.validation.check.check_finite [ref.func]
2024-11-21T19:28:52.6501769Z /home/runner/work/pyvista/pyvista/pyvista/core/_validation/validate.py:docstring of pyvista.core._validation.validate.validate_array:72: WARNING: py:func reference target not found: pyvista.core.validation.check.check_real [ref.func]
2024-11-21T19:28:52.6503851Z /home/runner/work/pyvista/pyvista/pyvista/core/_validation/validate.py:docstring of pyvista.core._validation.validate.validate_array:77: WARNING: py:func reference target not found: pyvista.core.validation.check.check_integer [ref.func]
2024-11-21T19:28:52.6505922Z /home/runner/work/pyvista/pyvista/pyvista/core/_validation/validate.py:docstring of pyvista.core._validation.validate.validate_array:82: WARNING: py:func reference target not found: pyvista.core.validation.check.check_sorted [ref.func]
2024-11-21T19:28:52.6507990Z /home/runner/work/pyvista/pyvista/pyvista/core/_validation/validate.py:docstring of pyvista.core._validation.validate.validate_array:95: WARNING: py:func reference target not found: pyvista.core.validation.check.check_range [ref.func]
2024-11-21T19:28:52.6521510Z /home/runner/work/pyvista/pyvista/pyvista/core/_validation/validate.py:docstring of pyvista.core._validation.validate.validate_transform3x3:18: WARNING: py:func reference target not found: pyvista.core.validation.check.check_finite [ref.func]
2024-11-21T19:28:52.6526255Z /home/runner/work/pyvista/pyvista/pyvista/core/_validation/validate.py:docstring of pyvista.core._validation.validate.validate_transform4x4:17: WARNING: py:func reference target not found: pyvista.core.validation.check.check_finite [ref.func]
2024-11-21T19:28:52.6528300Z /home/runner/work/pyvista/pyvista/doc/source/api/core/_validation.rst:18: WARNING: py:func reference target not found: pyvista.core._validation.validate_array3 [ref.func]
2024-11-21T19:28:52.6529877Z /home/runner/work/pyvista/pyvista/doc/source/api/core/_validation.rst:20: WARNING: py:func reference target not found: pyvista.core._validation.validate_arrayNx3 [ref.func]
2024-11-21T19:28:52.6531482Z /home/runner/work/pyvista/pyvista/doc/source/api/core/_validation.rst:22: WARNING: py:func reference target not found: pyvista.core._validation.validate_arrayN_unsigned [ref.func]
2024-11-21T19:28:52.6533088Z /home/runner/work/pyvista/pyvista/doc/source/api/core/_validation.rst:24: WARNING: py:func reference target not found: pyvista.core._validation.validate_transform4x4 [ref.func]

```

### CODE REVIEW

**Review by banesullivan — APPROVED** [review]

**Comment by user27182** [comment]
I haven't reviewed this closely but the changes from this PR, #6897 , and #6899 have all been patched into #6895, and it looks like a handful of the "fixes" didn't work. See https://github.com/pyvista/pyvista/actions/runs/12093521002/job/33724595350?pr=6895#step:12:1 for a list of currently outstanding broken links.

Rather than go through these "fix" PRs line by line (and manually review the docs) I think it's easier to apply a separate follow-up patch based on the `nitpicky` output after merging. But, this might add a small amount of unnecessary churn in the commits since a few lines will need to be modified again. 

I'll leave it up to reviewers to decide if it's better to make some changes here to have 100% fix rate with this PR or if it's ok to have 95% fixed, and fix the rest in a follow-up.

**Comment by user27182** [comment]
Alternatively, we can wait until #6895 is passing, then patch all changes here and make all the fixes in one go. In this case, since all the changes are made with `nitpicky=True` and passing, this should make sure we have a 100% fix rate. We can then merge this PR just before enabling `nitpicky` with #6895

**Comment by user27182** [comment]
The fixes here were patched in [9a78609](https://github.com/pyvista/pyvista/pull/6896/commits/9a78609a0558f79ebdaffa5e8986620560682f17) to include all remaining fixes from #6895.

**Review by banesullivan — APPROVED** [review]

**Review by tkoyama010 — APPROVED** [review]

**Pull request merged**
